fdbserver/storageserver.actor.cpp

```diff
@@ -3301,6 +3300,7 @@ ACTOR Future<bool> asyncPrepareVersionsForCommit_impl(StorageServerDisk* self, S
 		// Set the new durable version as part of the outstanding change set, before commit
 		data->storage.makeVersionDurable( newOldestVersion );
 	}
+	debug_advanceMaxCommittedVersion(data->thisServerID, newOldestVersion);
```

Comment:
@satherton Hey Steve, can you confirm this change is correct from the perspective of durability debugging?

Reply:
I'm not completely sure, the contract of the debug / version checking stuff isn't documented, but this looks right.  Advancing the max possible committed version should certainly be done before the commit begins, and in the event that the commit subsequently does not begin due to another failure that should be no different from a durability perspective than the commit beginning and then never completing, so this location seems like a good place to advance the version.